services/movie_session.py

```diff
@@ -0,0 +1,43 @@
+from db.models import MovieSession, Movie, CinemaHall
+
+
+def create_movie_session(movie_show_time: str,
+                         movie_id: int,
+                         cinema_hall_id: int) -> MovieSession:
+    movie = Movie.objects.get(id=movie_id)
+    cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+    return MovieSession.objects.create(show_time=movie_show_time,
+                                       movie=movie,
+                                       cinema_hall=cinema_hall)
+
+
+def get_movies_sessions(session_date: str = None) -> list[MovieSession]:
+    if session_date:
+        return MovieSession.objects.filter(show_time__date=session_date)
+    else:
+        return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(session_id: int,
+                         show_time: str = None,
+                         movie_id: int = None,
+                         cinema_hall_id: int = None) -> MovieSession:
+    session = MovieSession.objects.get(id=session_id)
+    if show_time:
+        session.show_time = show_time
+    if movie_id:
+        session.movie = Movie.objects.get(id=movie_id)
+    if cinema_hall_id:
+        session.cinema_hall = CinemaHall.objects.get(id=cinema_hall_id)
+
+    session.save()
+    return session
+
+
+def delete_movie_session_by_id(session_id: int) -> None:
+    session = MovieSession.objects.get(id=session_id)

```

Comment:
The `delete_movie_session_by_id` function does not handle the case where a movie session with the given ID does not exist. Consider using a try-except block to handle `MovieSession.DoesNotExist` exceptions.